media-video/obs-studio/obs-studio-27.2.0.ebuild

```diff
@@ -0,0 +1,201 @@
+
+EAPI=8
+
+CMAKE_REMOVE_MODULES_LIST=( FindFreetype )
+LUA_COMPAT=( luajit )
+PYTHON_COMPAT=( python3_{8..10} )
+
+inherit cmake lua-single python-single-r1 xdg
+
+OBS_BROWSER_COMMIT="b0d687937af876b52b69c46e276b4ab601b07f0e"
+CEF_DIR="cef_binary_4638_linux64"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/obsproject/obs-studio.git"
+	EGIT_SUBMODULES=( plugins/obs-browser )
+else
+	SRC_URI="https://github.com/obsproject/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	SRC_URI+=" browser? ( https://github.com/obsproject/obs-browser/archive/${OBS_BROWSER_COMMIT}.tar.gz -> obs-browser-${OBS_BROWSER_COMMIT}.tar.gz )"
+	KEYWORDS="~amd64 ~ppc64 ~x86"
+fi
+SRC_URI+=" browser? ( https://cdn-fastly.obsproject.com/downloads/${CEF_DIR}.tar.bz2 )"
+
+DESCRIPTION="Software for Recording and Streaming Live Video Content"
+HOMEPAGE="https://obsproject.com"
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="
+	+alsa browser decklink fdk jack lua nvenc pipewire
+	pulseaudio python speex +ssl truetype v4l vlc wayland
+"
+REQUIRED_USE="
+	browser? ( || ( alsa pulseaudio ) )
+	lua? ( ${LUA_REQUIRED_USE} )
+	python? ( ${PYTHON_REQUIRED_USE} )
+"
+
+BDEPEND="
+	lua? ( dev-lang/swig )
+	python? ( dev-lang/swig )
+"
+DEPEND="
+	dev-libs/glib:2
+	dev-libs/jansson:=
+	dev-qt/qtcore:5
+	dev-qt/qtdeclarative:5
+	dev-qt/qtgui:5[wayland?]
+	dev-qt/qtmultimedia:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtquickcontrols:5
+	dev-qt/qtsql:5
+	dev-qt/qtsvg:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtx11extras:5
+	dev-qt/qtxml:5
+	media-libs/libglvnd
+	media-libs/x264:=
+	media-video/ffmpeg:=[x264]
+	net-misc/curl
+	sys-apps/dbus
+	sys-libs/zlib:=
+	virtual/udev
+	x11-libs/libX11
+	x11-libs/libXcomposite
+	x11-libs/libXfixes
+	x11-libs/libXinerama
+	x11-libs/libXrandr
+	x11-libs/libxcb:=
+	alsa? ( media-libs/alsa-lib )
+	browser? (
+		app-accessibility/at-spi2-atk
+		app-accessibility/at-spi2-core:2
+		dev-libs/atk
+		dev-libs/expat
+		dev-libs/glib
+		dev-libs/nspr
+		dev-libs/nss
+		media-libs/alsa-lib
+		media-libs/fontconfig
+		media-libs/mesa[gbm(+)]
+		net-print/cups
+		x11-libs/libdrm
+		x11-libs/libXScrnSaver
+		x11-libs/libXcursor
+		x11-libs/libXdamage
+		x11-libs/libXext
+		x11-libs/libXi
+		x11-libs/libXrender
+		x11-libs/libXtst
+	)
+	fdk? ( media-libs/fdk-aac:= )
+	jack? ( virtual/jack )
+	lua? ( ${LUA_DEPS} )
+	nvenc? ( >=media-video/ffmpeg-4[video_cards_nvidia] )
+	pipewire? ( media-video/pipewire:= )
+	pulseaudio? ( media-sound/pulseaudio )
+	python? ( ${PYTHON_DEPS} )
+	speex? ( media-libs/speexdsp )
+	ssl? ( net-libs/mbedtls:= )
+	truetype? (
+		media-libs/fontconfig
+		media-libs/freetype
+	)
+	v4l? ( media-libs/libv4l )
+	vlc? ( media-video/vlc:= )
+	wayland? ( dev-libs/wayland )
+"
+RDEPEND="${DEPEND}"
+
+QA_PREBUILT="
+	usr/lib*/obs-plugins/chrome-sandbox
+	usr/lib*/obs-plugins/libEGL.so
+	usr/lib*/obs-plugins/libGLESv2.so
+	usr/lib*/obs-plugins/libcef.so
+	usr/lib*/obs-plugins/swiftshader/libEGL.so
+	usr/lib*/obs-plugins/swiftshader/libGLESv2.so
+"
```

Comment:
Hmmmm, I'm not getting this QA Notice?

In any case, added the files to the list (and re-sorted the list a bit).

Reply:
Did you setup flags check? Doesn't show by default.

For LDFLAGS, need `LDFLAGS="${LDFLAGS} -Wl,--defsym=__gentoo_check_ldflags__=0"`

For CFLAGS add `-frecord-gcc-switches` to CFLAGS, CXXFLAGS, FFLAGS, and FCFLAGS (need all 4 even for non-fortran because the shell script checks if the variable is set in each to activate, see /usr/lib/portage/python*/install-qa-check.d/10ignored-flags).

Then it should show with USE=browser assuming QA_PREBUILT is missing files

Reply:
I do remember setting some things in the past, but as I was looking into it earlier, I only saw 'qa' on 'PORTAGE_ELOG_CLASSES'.

Is `-Wl,--defsym=__gentoo_check_ldflags__=0` documented somewhere?  I think I've seen it before, but don't remember where.  A quick DDG does lead me to the arch testers wiki article, which does mention it [1].

1. https://wiki.gentoo.org/wiki/Project:AMD64_Arch_Testers

Reply:
Adding qa is good to at least see QA notice at all (well they still show in build log either way but not reported after the build otherwise).

*FLAGS is indeed not well documented as far as I'm aware (not that I checked much, maybe there's some page), although recent'ish bugs from ago try to give instructions for confused maintainers:
```
NOTE:
To reproduce this issue you may want to add "-frecord-gcc-switches" to
CFLAGS/CXXFLAGS/FFLAGS/FCFLAGS. If this is not something about
c/c++ context see this bug as an hint to hide the QA warning (with
QA_FLAGS_IGNORED) where is not possible to respect CFLAGS.
```
Think most people don't really check these and it tend to fall back to tinderboxes.

Reply:
https://wiki.gentoo.org/wiki/Package_testing#Configuration